test/tests/hostmask.js

```diff
@@ -60,3 +67,38 @@ describe("Hostmask", function () {
 		expect(Helper.compareHostmask(a, b)).to.be.true;
 	});
 });
+
+describe("compareWithWildcard", function () {
+	const goodPairs = [
+		["asdf", "asdf"],
+		["a?df*", "asdf"],
+		["*asdf*", "asdf"],
+		["*asdf", "asdf"],
+		["asd?", "asdf"],
+		["asd?*", "asdf"],
```

Comment:
Could you add these cases?

```js
["AsDf", "asdf"],
["a??f", "asdf"],
["a*", "asdf"],
["*f", "asdf"],
["*s*", "asdf"],
```

If `String.split()` worked like it did in Python, the last 4 cases would've failed. Thought I'd found a bug at first!

Reply:
what do you mean?
```python
>>> "*".join("*s*".split("*"))
'*s*'
```

Reply:
Oh! I just got mixed up.

`split` without arguments will collapse empty splits but apparently not if you give a delimiter explicitly.